backend/sirius-web-spring-collaborative-diagrams/src/main/java/org/eclipse/sirius/web/spring/collaborative/diagrams/handlers/UpdateNodePositionEventHandler.java

```diff
@@ -0,0 +1,119 @@
+import org.eclipse.sirius.web.collaborative.diagrams.api.IDiagramService;
+import org.eclipse.sirius.web.collaborative.diagrams.api.dto.UpdateNodePositionInput;
+import org.eclipse.sirius.web.collaborative.diagrams.api.dto.UpdateNodePositionSuccessPayload;
+import org.eclipse.sirius.web.core.api.ErrorPayload;
+import org.eclipse.sirius.web.core.api.IEditingContext;
+import org.eclipse.sirius.web.diagrams.MoveEvent;
+import org.eclipse.sirius.web.diagrams.Node;
+import org.eclipse.sirius.web.diagrams.Position;
+import org.eclipse.sirius.web.spring.collaborative.diagrams.messages.ICollaborativeDiagramMessageService;
+import org.springframework.stereotype.Service;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Handle "Update Node Position" events.
+ *
+ * @author fbarbin
+ */
+@Service
+public class UpdateNodePositionEventHandler implements IDiagramEventHandler {
+
+    private final ICollaborativeDiagramMessageService messageService;
+
+    private final IDiagramService diagramService;
+
+    private final Counter counter;
+
+    public UpdateNodePositionEventHandler(ICollaborativeDiagramMessageService messageService, IDiagramService diagramService, MeterRegistry meterRegistry) {
+        this.diagramService = Objects.requireNonNull(diagramService);
+        this.messageService = Objects.requireNonNull(messageService);
+
+        // @formatter:off
+        this.counter = Counter.builder(Monitoring.EVENT_HANDLER)
+                .tag(Monitoring.NAME, this.getClass().getSimpleName())
+                .register(meterRegistry);
+        // @formatter:on
+    }
+
+    @Override
+    public boolean canHandle(IDiagramInput diagramInput) {
+        return diagramInput instanceof UpdateNodePositionInput;
+    }
+
+    @Override
+    public EventHandlerResponse handle(IEditingContext editingContext, IDiagramContext diagramContext, IDiagramInput diagramInput) {
+        this.counter.increment();
+
+        EventHandlerResponse result;
+        if (diagramInput instanceof UpdateNodePositionInput) {
+            result = this.handleUpdateNodePosition(editingContext, diagramContext, (UpdateNodePositionInput) diagramInput);
+        } else {
+            String message = this.messageService.invalidInput(diagramInput.getClass().getSimpleName(), UpdateNodePositionEventHandler.class.getSimpleName());
+            result = new EventHandlerResponse(false, representation -> false, new ErrorPayload(message));
+        }
+        return result;
+    }
+
+    private EventHandlerResponse handleUpdateNodePosition(IEditingContext editingContext, IDiagramContext diagramContext, UpdateNodePositionInput diagramInput) {
+        // @formatter:off
+        Position newPosition = Position.newPosition()
+                .x(diagramInput.getNewPositionX())
+                .y(diagramInput.getNewPositionY())
+                .build();
+        // @formatter:on
```

Comment:
This pattern feels really overkill every time I read it. I'd propose `Position.at(x, y)` as a simpler, shorter and more readable factory method. @sbegaudeau wdyt?

Reply:
it could be simplified indeed, maybe later in order to move forward with this review but I agree